main.go

```diff
@@ -61,62 +53,169 @@ func List() {
 	Sort(entries)
 	Header()
 	for i, j := range entries {
-		fmt.Printf("%4d %14s %14s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
+		fmt.Printf("%10d %20s %20s %20s\n", j.ID, j.LastName, j.FirstName, j.PhoneNumber)
 		if (i+1)%20 == 0 && i < len(entries) {
+			fmt.Println(" ")
 			fmt.Println("Press <ENTER> to continue...")
+			fmt.Println(" ")
 			fmt.Scanln()
 			Header()
 		}
 	}
+}
 
+func GiveID(entries []Entry) (i uint) {
+	i = uint(len(entries)) + 1 //Thats not ideal and will work wrong if the last contact will be deleted
```

Comment:
Yup, try to fix it. It should be a hybrid between your previous and this current solutions